tests/TestCase/Collection/CollectionTest.php

```diff
@@ -2553,30 +2552,6 @@ protected function datePeriod($start, $end): DatePeriod
         return new \DatePeriod(new \DateTime($start), new \DateInterval('P1D'), new \DateTime($end));
     }
 
-    /**
-     * Tests to ensure that collection classes extending ArrayIterator work as expected.
-     */
-    public function testArrayIteratorExtend(): void
-    {
-        $iterator = new TestIterator(range(0, 10));
-
-        $this->assertTrue(method_exists($iterator, 'checkValues'));
-        $this->assertTrue($iterator->checkValues());
-
-        //We need to perform at least two collection operation to trigger the issue.
-        $newIterator = $iterator
-            ->filter(function ($item) {
-                return $item < 5;
-            })
-            ->reject(function ($item) {
-                return $item > 2;
-            });
```

Comment:
This was a bug/regression from a few years back. Removing the test means we *could* regress here again, but I suspect this is unlikely to happen unless we were to make dramatic changes to `CollectionTrait`.

Reply:
Well, CollectionTrait::append() and ArrayIterator::append() are conflicting. I'm not sure what users were trying to do with that.